### ISSUE TITLE
Code Sample Fails

### ISSUE THREAD

**Issue by jeff-hykin** [parcel-bundler/parcel#6887]
# 🐛 bug report

- copy paste an example from the website
- create the supporting files (`a.js`)
- run it
- get error

## 🎛 Configuration (.babelrc, package.json, cli command)

None/default

## 🤔 Expected Behavior

Not error

## 😯 Current Behavior

```
> err is: TypeError [ERR_INVALID_ARG_TYPE]: The "to" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:329:5)
    at validateString (node:internal/validators:129:11)
    at Object.relative (node:path:1096:5)
    at processConfig (/Users/jeffhykin/node_modules/@parcel/core/lib/loadParcelConfig.js:86:36)
    at create (/Users/jeffhykin/node_modules/@parcel/core/lib/loadParcelConfig.js:71:10)
    at loadParcelConfig (/Users/jeffhykin/node_modules/@parcel/core/lib/loadParcelConfig.js:48:26)
    at async Parcel.init (/Users/jeffhykin/node_modules/@parcel/core/lib/Parcel.js:198:9)
    at async Parcel.run (/Users/jeffhykin/node_modules/@parcel/core/lib/Parcel.js:250:7)
    at async REPL17:3:40 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

## 💁 Possible Solution

I looked into the source, its erroring because a filepath is null, but I'm not sure why its null or why its being called.

## 🔦 Context

I'm trying to convert from parcel 1's express middleware setup and having a hard time.

## 💻 Code Sample

The "Build" code sample from here: https://v2.parceljs.org/features/parcel-api/
Converted to work inside a node repl.
Only file in the project was `a.js`

```js
Parcel  = require("@parcel/core").default

bundler = new Parcel({
    entries: 'a.js',
    defaultConfig: '@parcel/config-default'
})

;((async ()=>{
    try {
        let {bundleGraph, buildTime} = await bundler.run();
        let bundles = bundleGraph.getBundles();
        console.log(`✨ Built ${bundles.length} bundles in ${buildTime}ms!`);
    } catch (err) {
        console.debug(`err is:`,err)
        console.log(err.diagnostics);
    }
})())
```

## 🌍 Your Environment

Using `node v15.14.0`, no config, other than `a.js` totally empty project, running inside node repl.
| Software         | Version(s) |
| ---------------- | ---------- |
| @parcel/core           | 2.0.0-alpha.3
| Node             | v15.14.0
| npm/Yarn         | 7.7.6
| Operating System | MacOS 10.13.6

**Comment by mischnic** [comment]
> @parcel/core | 2.0.0-alpha.3

You should use the RC versions of core and config-default

**Comment by jeff-hykin** [comment]
> > @parcel/core | 2.0.0-alpha.3
> 
> You should use the RC versions of core and config-default
> 
> 
> 

How would I do that? @mischnic 
Like `npm install @parcel/core@???` and 

```js
bundler = new Parcel({
    entries: 'a.js',
    defaultConfig: '@parcel/config-default'
              // ^ what here 
})

```

**Comment by mischnic** [comment]
Leave your code as it is and use `npm install @parcel/core@next @parcel/config-default@next` instead

**Comment by jeff-hykin** [comment]
Sorry for the delay, had a few things to take care of. Using @next solved the core problem, but I still had a few issues and think the docs need to be updated to cover them @mischnic 

TLDR:
- I went ahead and created a PR [here](https://github.com/parcel-bundler/website/pull/932) to try to cover this situation.
- Probably should add a page for the broken link (https://v2.parceljs.org/configuration/package-json), or remove the link from the error message. But maybe thats already in the roadmap. 


Running with @next got me to this error

- ![Screen Shot 2021-09-11 at 8 03 57 AM](https://user-images.githubusercontent.com/17692058/132948782-24be451d-f29d-43c3-a665-60839c51e80a.png)

- Digging into it a bit more gets me a link to a page that doesn't exist (https://v2.parceljs.org/configuration/package-json)

- ![Screen Shot 2021-09-11 at 8 10 35 AM](https://user-images.githubusercontent.com/17692058/132948941-c108d8ca-d238-4cb6-a7ed-a93af1c4628f.png)

- I didn't know what "main" was since I wasn't providing a main option in node. Thankfully that broken link was enough of a hint for me to look at the package json, which I had simply generated with `npm init -y`. If the link had worked, I'd probably consider the issue closed cause that was a really helpful hint/explaination.

- Changing the "main" in the package json to something other than "a.js" worked.
